addons/web_editor/static/src/js/wysiwyg/widgets/media.js

```diff
@@ -584,15 +612,42 @@ var FileWidget = SearchableMediaWidget.extend({
                             'res_model': self.options.res_model,
                             'width': 0,
                             'quality': 0,
+                        }}, {
+                            xhr: function () {
+                                var xhr = $.ajaxSettings.xhr();
+                                xhr.upload.onprogress = function (ev) {
+                                    var prcComplete = ev.loaded / ev.total * 100;
+                                    $progressBar.find('.progress-bar').css({
+                                        width: parseInt(prcComplete) + '%',
+                                    }).text(prcComplete.toFixed(2) + '%');
+                                };
+                                xhr.upload.onload = function () {
+                                    // Don't show yet success as backend code only starts now
+                                    $progressBar.find('.progress-bar').css({width: '100%'}).text('100%');
+                                };
+                                return xhr;
+                            },
                         },
-                    }).then(function (attachment) {
+                    ).then(function (attachment) {
+                        $progressBar.find('.fa-spinner, .progress').addClass('d-none');
+                        $progressBar.find('.js_progressbar_txt .text-success').removeClass('d-none');
                         self._handleNewAttachment(attachment);
+                    }).guardedCatch(function () {
+                        hasError = true;
+                        $progressBar.find('.fa-spinner, .progress').addClass('d-none');
+                        $progressBar.find('.js_progressbar_txt .text-danger').removeClass('d-none');
                     });
                 });
             });
         });
 
         return uploadMutex.getUnlockedDef().then(function () {
+            if (!hasError) {
+                setTimeout(() => {
+                    // Don't call notification.close as parent is destroyed in single image upload
```

Comment:
If parent is destroyed, the related notifications are not automatically closed? :fearful:

Reply:
Done, see `wait` param